fv3core/stencils/fxadv.py

```diff
@@ -387,33 +377,33 @@ def fxadv_fluxes_stencil(
             yfx_adv = dx * prod * sin_sg4[0, -1] if prod > 0 else dx * prod * sin_sg2
 
 
-@gtstencil()
-def flux_divergence_area(
-    area: FloatFieldIJ,
-    xfx_adv: FloatField,
-    yfx_adv: FloatField,
-    ra_x: FloatField,
-    ra_y: FloatField,
-):
-    """Compute the area with flux divergence applied
-     Args:
-       xfx_adv: Finite volume flux form operator in x direction (in)
-       yfx_adv: Finite volume flux form operator in y direction (in)
-       ra_x: Area increased in the x direction due to flux divergence (inout)
-       ra_y: Area increased in the y direction due to flux divergence (inout)
-    Grid variable inputs:
-       area
+@gtscript.function
+def apply_x_flux_divergence(q: FloatField, q_x_flux: FloatField) -> FloatField:
     """
-    from __externals__ import local_ie, local_is, local_je, local_js
-
-    with computation(PARALLEL), interval(...):
-        with horizontal(region[local_is : local_ie + 2, :]):
-            ra_x = ra_x_func(area, xfx_adv)
-        with horizontal(region[:, local_js : local_je + 2]):
-            ra_y = ra_y_func(area, yfx_adv)
+    Update a scalar q according to its flux in the x direction.
+    """
+    return q + q_x_flux - q_x_flux[1, 0, 0]
 
 
-def compute(uc, vc, crx_adv, cry_adv, xfx_adv, yfx_adv, ut, vt, ra_x, ra_y, dt):
+@gtscript.function
+def apply_y_flux_divergence(q: FloatField, q_y_flux: FloatField) -> FloatField:
+    """
+    Update a scalar q according to its flux in the x direction.
+    """
+    return q + q_y_flux - q_y_flux[0, 1, 0]
+
+
+def compute(
+    uc,
+    vc,
+    crx_adv,
+    cry_adv,
+    xfx_adv,
+    yfx_adv,
```

Comment:
this seems to be the only place now where it is called xfx_adv, yfx_adv .. though I probably don't want you to change it because it'll call more merge conflicts for the single stencil version waiting in the wings